### PR TITLE
chore(gatsby-transformer-remark): add pluginOptionsSchema export

### PR DESCRIPTION
**Pull request by mfrachet**
## Description

Add plugin schema options in `gatsby-transformer-remark`

Relates to https://github.com/gatsbyjs/gatsby/pull/27242 and https://www.gatsbyjs.com/plugins/gatsby-transformer-remark/

### CODE REVIEW

File: packages/gatsby-transformer-remark/src/gatsby-node.js [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+  exports.pluginOptionsSchema = function ({ Joi }) {
+    return Joi.object({
+      commonmark: Joi.boolean().description(
+        `Activates CommonMark mode (default: true)`
+      ),
+      footnotes: Joi.boolean().description(
+        `Activates Footnotes mode (default: true)`
+      ),
+      pedantic: Joi.boolean().description(
+        `Activates Pandemic mode (default: true)`
```
Comment (mxstbr) [inline packages/gatsby-transformer-remark/src/gatsby-node.js:15]:
```suggestion
        `Activates pedantic mode (default: true)`
```

**Review by mxstbr — APPROVED** [review]
One tiny typo, otherwise this looks great to me!!

File: packages/gatsby-transformer-remark/src/gatsby-node.js (hunk 1 above)
Comment (mfrachet) [reply packages/gatsby-transformer-remark/src/gatsby-node.js:15]: Ooh wow... COVID is hitting hard 🤣

**Review by mxstbr — APPROVED** [review]

File: packages/gatsby-transformer-remark/src/gatsby-node.js [hunk 2]
```diff
@@ -1,3 +1,33 @@
 exports.createSchemaCustomization = require(`./create-schema-customization`)
 exports.onCreateNode = require(`./on-node-create`)
 exports.setFieldsOnGraphQLNodeType = require(`./extend-node-type`)
+
+if (process.env.GATSBY_EXPERIMENTAL_PLUGIN_OPTION_VALIDATION) {
+  exports.pluginOptionsSchema = function ({ Joi }) {
+    return Joi.object({
+      commonmark: Joi.boolean().description(
+        `Activates CommonMark mode (default: true)`
+      ),
```
Comment (muescha) [inline packages/gatsby-transformer-remark/src/gatsby-node.js:10]: is here a missing default value? `.boolean().default(true)`

**Pull request merged**